src/irreps/anirrep.jl

```diff
@@ -0,0 +1,304 @@
+    L = length(v)
+    @boundscheck 1 <= i <= L || throw(BoundsError(v, i))
+    return ANIrrep{N}(i - 1)
+end
+
+findindex(::SectorValues{ANIrrep{N}}, a::ANIrrep{N}) where {N} = a.n + 1
+
+# Product iterator
+# ----------------
+
+const ANIrrepProdIterator{N} = SectorProductIterator{ANIrrep{N}}
+⊗(a::ANIrrep{N}, b::ANIrrep{N}) where {N} = SectorProductIterator((a <= b ? (a, b) : (b, a))...)
+
+Base.IteratorSize(::Type{ANIrrepProdIterator{N}}) where {N} = Base.HasLength()
+function Base.length(x::ANIrrepProdIterator{N}) where {N}
+    N < 4 && return 1 # abelian
+    a, b = x.a, x.b
+    if a == b
+        return a == 3 ? 5 : 1 # 3 x 3 = 1 + 1' + 1'' + 2*3
+    else
+        return 1
+    end
+end
+
+# TODO: any way to shorten this?
+function Base.iterate(p::ANIrrepProdIterator{N}, state::Int = 1) where {N}
+    a, b = p.a, p.b
+    if N < 4
+        return state > 1 ? nothing : (ANIrrep{N}((a.n + b.n) % _npartitions(N)), state + 1)
+    else
+        u, u′, u′′, three = ANIrrep{4}(0), ANIrrep{4}(1), ANIrrep{4}(2), ANIrrep{4}(3)
+        if state == 1
+            # rules with the trivial irrep
+            a.n == 0 && return (b, 2)
+            b.n == 0 && return (a, 2)
+            # 1' rules
+            if a.n + b.n == 3 # (1, 2) or (2, 1)
+                return (u, 2)
+            elseif a.n == b.n != 3 # (1,1) or (2,2)
+                return (A4Irrep((2 * a.n) % 3), 2)
+            elseif (a.n in (1, 2) && b.n == 3) || (a.n == 3 && b.n in (1, 2))
+                # all 1D × 3 = 3
+                return (three, 2)
+            else # 3 x 3
+                return (u, 2) # first in sequence
+            end
+        elseif state == 2 && a.n == 3 && b.n == 3
+            return (u′, 3)
+        elseif state == 3 && a.n == 3 && b.n == 3
+            return (u′′, 4)
+        elseif state == 4 && a.n == 3 && b.n == 3
+            return (three, 5)
+        else
+            return nothing
+        end
+    end
+end
+
+# Topological data
+# ----------------
+function dim(a::ANIrrep{N}) where {N}
+    N < 4 && return 1
+    return a.n == 3 ? 3 : 1
+end
+
+function Nsymbol(a::ANIrrep{N}, b::ANIrrep{N}, c::ANIrrep{N}) where {N}
+    N < 4 && return (c.n == (a.n + b.n) % _npartitions(N)) ? 1 : 0
+    # N = 4
+    u, u′, u′′, three = ANIrrep{4}(0), ANIrrep{4}(1), ANIrrep{4}(2), ANIrrep{4}(3)
+    if a == three
+        if b == three # 3 x 3
+            return (c == u || c == u′ || c == u′′) ? 1 : 2
+        else # 3 x 1D
+            return c == three ? 1 : 0
+        end
+    else
+        if b == three # 1D x 3
+            return c == three ? 1 : 0
+        else # 1D x 1D
+            return c.n == (a.n + b.n) % 3 ? 1 : 0
+        end
+    end
+end
+
+function Fsymbol(a::I, b::I, c::I, d::I, e::I, f::I) where {N, I <: ANIrrep{N}}
+    T = sectorscalartype(I)
+    Nabe = Nsymbol(a, b, e)
+    Necd = Nsymbol(e, c, d)
+    Nbcf = Nsymbol(b, c, f)
+    Nafd = Nsymbol(a, f, d)
+
```

Comment:
Probably have to special case `N < 4`, for `UniqueFusion` the result has to be a scalar, not a matrix